### ISSUE TITLE
All commits added to changelog after merging a branch containing a release

### ISSUE THREAD

**Issue by loderunner** [googleapis/release-please#2044]
#### TL;DR

After making a release with release-please on a "hotfix" emergency branch, I rebased on `main` and squash merged the hotfix branch to `main`. After creating a release pr on `main`, all commits since the beginning are added to the changelog.

#### Expected behavior

The changelog contains only the commits since latest ("hotfix") release.

####Observed behavior

The changelog contains all commits since the beginning of the repo.

#### Example repository

https://github.com/loderunner/devexp-flow

* the 1.1.1 release is the hotfix release
* the 1.2.0 release PR is the next trunk release
* the [1.2.0 release PR changelog](https://github.com/loderunner/devexp-flow/pull/15) shows all commits since the beginning
* [comparing 1.1.1 to `main`](https://github.com/loderunner/devexp-flow/compare/1.1.1...main) only yields 3 commits, that are the ones I expect to see in my changelog

#### Environment details

  - OS: Ubuntu 20.04.6 LTS
  - Node.js version: 20.2.0
  - npm version: 9.6.6
  - `release-please` version: 15.12.0

#### Steps to reproduce

1. Initialize `release-please` on a node project with `release-please bootstrap --type node`
2. Add a feature commit on `main`, `git commit -m 'feat: initial commit'`
3. Create a GitHub release on `main` with `release-please github-release --target-branch main`
4. Add a feature commit on `main`, `git commit -m 'feat: some feature'`
5. Create a release PR on `main` with `release-please release-pr --target-branch main`
6. Create a `hotfix` branch based on latest release (before "some feature") `git switch -c hotfix 1.0.0`
7. Add a fix commit on the `hotfix` branch, `git commit -m 'fix: some fix'`
8. Create a release PR on the hotfix branch `release-please release-pr --target-branch hotfix`
9. Squash merge the release PR into the `hotfix` branch
10. Create a release from the `hotfix` branch, `release-please github-release --target-branch hotfix`
11. Rebase the `hotfix` branch on `main`, `git rebase main hotfix`
12. Open a PR from `hotfix` to `main` and squash merge it (commit message 'fix: some fix')
13. Update the release PR on `main`, `release-please release-pr --target-branch main`

If I didn't get this wrong, the release PR changelog _should_ include `feat: initial commit`. Feel free to ask me follow-up questions as this scenario is drawn from memory and might not work.

**Comment by loderunner** [comment]
From my command-line:

```
❯ git log --oneline 1.1.1..main
c3e7c06 (HEAD -> main, origin/main) fix: remove comma (#13)
5b99a3b feat: add random number at the end (#11)
f6a5870 fix: use ISO timestamp (#9)
```

release-please debug output:

```
❯ npx release-please release-pr --dry-run --token $GITHUB_TOKEN --repo-url loderunner/devexp-flow --target-branch main --debug --config-file .circleci/release/release-please-config.json --manifest-file .circleci/release/release-please-manifest.json
❯ Fetching .circleci/release/release-please-config.json from branch main
❯ Fetching .circleci/release/release-please-manifest.json from branch main
✔ Building pull requests
✔ Building strategies by path
❯ .: node
✔ Collecting release commit SHAs
❯ release search depth: 400
❯ Fetching releases with cursor undefined
❯ Found release for path ., 1.1.1
❯ release for path: ., version: 1.1.1, sha: bf11e284392e376af801edb1b6067e0132870086
✔ Collecting commits since all latest releases
❯ commit search depth: 500
❯ Set(1) { 'bf11e284392e376af801edb1b6067e0132870086' }
❯ Fetching merge commits on branch main with cursor: undefined
❯ Backfilling file list for commit: 5dd8a79c0def3bb85ec77f0c3c6d23fef5d3c14d
❯ Found 1 files
❯ Backfilling file list for commit: cbc01fc11b88607dfdef0a31c7301ca08733f025
❯ Found 1 files
❯ Backfilling file list for commit: b4a8c5acd9242c50bbb6115c2bf5b15edbc7000b
❯ Found 1 files
❯ Backfilling file list for commit: c79143bdb2e51a1e72abdb0246f2a9d96f1ccbfd
❯ Found 1 files
❯ Backfilling file list for commit: b9c12e28900a5dbcd9577b42f686d348d2fc34db
❯ Found 1 files
❯ Backfilling file list for commit: a7d45f305842b5dd7bea956206de06410914fad2
❯ Found 1 files
❯ Backfilling file list for commit: ddb8356c573a3cb38452c8ef49d42ec92df263c2
❯ Found 2 files
⚠ Expected 1 commits, only found 0
✔ Splitting 18 commits by path
✔ Building candidate release pull request for path: .
❯ type: node
❯ targetBranch: main
❯ commit could not be parsed: b9c12e28900a5dbcd9577b42f686d348d2fc34db add gitignore
❯ commit could not be parsed: a7d45f305842b5dd7bea956206de06410914fad2 add some code
❯ commit could not be parsed: ddb8356c573a3cb38452c8ef49d42ec92df263c2 initial commit
❯ commits: 15
⚠ No latest release pull request found.
✔ Considering: 15 commits
❯ component:
❯ pull request title pattern: undefined
❯ Fetching package.json from branch main
Would open 1 pull requests
fork: false
title: chore(main): release 1.2.0
branch: release-please--branches--main--components--devexp-flow
draft: false
body: :robot: I have created a release *beep* *boop*
---


## [1.2.0](https://github.com/loderunner/devexp-flow/compare/1.1.1...1.2.0) (2023-08-11)


### Features

* add a timestamp to the message ([#7](https://github.com/loderunner/devexp-flow/issues/7)) ([2cf08f2](https://github.com/loderunner/devexp-flow/commit/2cf08f2c5b2c22923ce08acb18404f6a9950b139))
* add random number at the end ([#11](https://github.com/loderunner/devexp-flow/issues/11)) ([5b99a3b](https://github.com/loderunner/devexp-flow/commit/5b99a3b4ee92825ed419d87a904d2f8452e95865))
* change hello message ([#5](https://github.com/loderunner/devexp-flow/issues/5)) ([6635db0](https://github.com/loderunner/devexp-flow/commit/6635db08e553a97be27e7d050c4591a247245aee))


### Bug Fixes

* remove comma ([#13](https://github.com/loderunner/devexp-flow/issues/13)) ([c3e7c06](https://github.com/loderunner/devexp-flow/commit/c3e7c065ae8fad5ab7f98fa86a1b26be607c6a86))
* use ISO timestamp ([#9](https://github.com/loderunner/devexp-flow/issues/9)) ([f6a5870](https://github.com/loderunner/devexp-flow/commit/f6a587085ed96bf2a319265c1a346dfbda9a9867))

---
This PR was generated with [Release Please](https://github.com/googleapis/release-please). See [documentation](https://github.com/googleapis/release-please#release-please).
updates: 7
❯ Fetching package-lock.json from branch main
✔ updating from 1.1.1 to 1.2.0
❯ Fetching npm-shrinkwrap.json from branch main
⚠ file npm-shrinkwrap.json did not exist
❯ Fetching samples/package.json from branch main
⚠ file samples/package.json did not exist
❯ Fetching CHANGELOG.md from branch main
❯ Fetching package.json from branch main
✔ updating from 1.1.1 to 1.2.0
❯ Fetching changelog.json from branch main
⚠ file changelog.json did not exist
❯ Fetching .circleci/release/release-please-manifest.json from branch main
  package-lock.json:  [class PackageLockJson extends DefaultUpdater]
  npm-shrinkwrap.json:  [class PackageLockJson extends DefaultUpdater]
  samples/package.json:  [class SamplesPackageJson extends DefaultUpdater]
  CHANGELOG.md:  [class Changelog extends DefaultUpdater]
  package.json:  [class PackageJson extends DefaultUpdater]
  changelog.json:  [class ChangelogJson extends DefaultUpdater]
  .circleci/release/release-please-manifest.json:  [class ReleasePleaseManifest extends DefaultUpdater]
```

**Comment by loderunner** [comment]
I think I just understood what happens here, sorry for creating a bug issue if I got this right:

release-please uses the manifest file to lookup the latest version, not the tag.

Still, is there a way for me to avoid including commits to the changelog that have been included in previous releases?

**Comment by arrrgi** [comment]
Can confirm I experience the same too. Each releases changelog notes are additive - all previous changes plus the latest changes are generated rather just the changes since the last release, then the same occurs for the release after that, adding all changes since 0.1.0.

**Comment by sam3k** [comment]
> I think I just understood what happens here, sorry for creating a bug issue if I got this right:
> 
> release-please uses the manifest file to lookup the latest version, not the tag.
> 
> Still, is there a way for me to avoid including commits to the changelog that have been included in previous releases?

I think this is the same issue I am having (using the Github action version). Isn't the manifest optional and primarily for repos hosting multiple libraries? See [this](https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md#manifest-driven-release-please).

The issues I am seeing are:
* Changelogs for latest version show all commits as mentioned here
* I forced pushed the version via `Release As` empty commit to version 0.4.1
* When run again, it says the latest version is 0.4.0 (it is not) and `this.versioningStrategy.bump()` then tries to bump to 0.4.1 (but it is already 0.4.1) 
* Release tag no longer generated

**Comment by furlangiacomo** [comment]
Hi, has anyone found a possible workaround or solution for this?

**Comment by mikavilpas** [comment]
I think I had a problem like this - but I don't have the details any longer and my memory is hazy...

I _think_ in my case I had some closed tickets with labels that had special meaning to release-please. I was able to resolve the issue by removing those labels and rerunning the job (helps to have this in the workflow)

```yaml
on:
  workflow_dispatch: # makes it possible to run the workflow manually

```

This might not be helpful but wanted to try anyway...
